web/client/components/map/leaflet/plugins/VectorLayer.jsx

```diff
@@ -40,16 +40,28 @@ var createVectorLayer = function(options) {
         style: options.nativeStyle || options.style || defaultStyle // TODO ol nativeStyle should not be taken from the store
     });
     layer.setOpacity = (opacity) => {
-        const style = assign({}, layer.options.style || defaultStyle, {opacity: opacity, fillOpacity: opacity});
+        const style = assign({}, layer.options.style || defaultStyle, { opacity: opacity, fillOpacity: opacity });
         layer.setStyle(style);
         setOpacity(layer, opacity);
     };
+    layer.on('layeradd', () => {
+        layer.setOpacity(layer.opacity || options.opacity);
+    });
     return layer;
 };
 
 Layers.registerType('vector', {
     create: (options) => {
-        return createVectorLayer(options);
+        const layer = createVectorLayer(options);
+        // layer.opacity will store the opacity value
+        // to be applied to layer style once the layer is ready
+        layer.opacity = options.opacity || 1.0;
```

Comment:
Same as above, 0 can be a value and in this case opacity will become 1. Please make check it works for 0 value.